addons/mail/static/src/models/thread/thread.js

```diff
@@ -1340,6 +1376,15 @@ function factory(dependencies) {
          * interface and to notify the server of the new state.
          */
         pendingFoldState: attr(),
+        /**
+         * Determine if there is a pending pin state change, which is a change
+         * of pin state requested by the client but not yet confirmed by the
+         * server.
+         *
+         * This field can be updated to immediately change the fold state on the
+         * interface and to notify the server of the new state.
+         */
+        pendingPinState: attr(),
```

Comment:
Should probably be boolean since this is the type of the original value. (and then condition on undefined vs false to be clarified)

Reply:
for those, I clearly mimicked what foldState holds: a string.
Since those 4 fields (pending/server[STATE]) manage the same technical use case, and are similar in nature, I prefer to have string metadata rather than a boolean

Reply:
It's not a big deal either way, but here is the reasoning:
fold_state is a string because it can have more than 2 values. (Selection field in python)
is_pinned is Boolean

Reply:
Agree with @seb-odoo, when there were 1 pin status fields, made sense to have string with 4 values. Now with 2 fields, they should be booleans (the pending one should either allow `undefined` or there should be a 3rd boolean field to determine if pending is in application).